pkg/instrumentation/sdk.go

```diff
@@ -395,6 +395,9 @@ func (i *sdkInjector) injectCommonSDKConfig(ctx context.Context, otelinst v1alph
 }
 
 func chooseServiceName(pod corev1.Pod, resources map[string]string, index int) string {
+	if name := chooseLabelOrAnnotation(pod, semconv.ServiceNameKey, constants.AnnotationAppName); name != "" {
+		return name
+	}
```

Comment:
I am concerned that this would be a breaking change.

Reply:
yes, I'll include that in the changelog

Reply:
@jaronoff97 @pavolloffay I am pretty worried about this breaking change.  Changes to `service.name` will change emitted telemetry, which could affect end-users alerts/SLOs/dashboards/etc. This change will (likely) not  be detected with unit or e2e tests before a user updates their operator. Additionally, the operator's auto-instrumentation injection is already a huge obfuscation on top of already obfuscated auto-instrumentation, which means that our end-users are even less likely to understand the impact of this change.

I would much rather prefer this work be implemented in such a way that the annotation is used as the service name if no other service name is set OR if some other explicit configuration field is set that enables this behavior.

Reply:
The reason I call this out as a breaking change in the current implementation is that it is now defaulting to the annotation value if it exists.  Since it is a common annotation, it will probably be on most end-user's pods, resulting in the potential breaking change.

I would much rather this be the last option in this list to avoid the breaking change.

Reply:
In addition, there is no stable semantic convention that ranks which value should be used as the `service.name`. I'd prefer avoiding breaking changes on this topic until there is a stable semantic convention, so that we can minimize the number of breaking changes.

Reply:
we could solve this problem if we let the user define the mapping they prefer - so it would be an opt-in feature

Reply:
from SIG:
I think we should move forward with this by putting it within a new "defaults" section in the instrumentation CRD as so:
```
apiVersion: opentelemetry.io/v1alpha1
kind: Instrumentation
metadata:
  name: my-instrumentation
spec:
  defaults:
    useLabelsForResource: true
```

We also must then document in the readme the precedence for the automatic variables we set for users and how this boolean changes that.